plasma_framework/contracts/src/framework/utils/Quarantine.sol

```diff
@@ -0,0 +1,44 @@
+pragma solidity ^0.5.0;
+
+import "../utils/Operated.sol";
+
+/**
+ * @title Provides a way to quarantine (disable) contracts for a period of time
+ */
+contract Quarantine is Operated {
+    mapping(address => uint256) private _quarantine;
+    uint256 private _quarantinePeriod;
+    uint8 private _immunitiesRemaining;
+
+    /**
+     * @dev The _initialImmuneCount parameter allows us to deploy the platform with some
+     * pre-verified contracts that don't get quarantined.
+     * @param _period the period of time in seconds to quarantine contracts.
+     * @param _initialImmuneCount the first _initialImmuneCount contracts don't get quarantined.
+     */
+    constructor (uint256 _period, uint8 _initialImmuneCount) public {
+        _quarantinePeriod = _period;
+        _immunitiesRemaining = _initialImmuneCount;
+    }
+
+    modifier notQuarantined(address _contractAddress) {
+        require(block.timestamp > _quarantine[_contractAddress], "Contract is quarantined.");
+        delete _quarantine[_contractAddress];
```

Comment:
A general best practice is not to have things aside from `assertion` in modifier.
https://consensys.github.io/smart-contract-best-practices/recommendations/#use-modifiers-only-for-assertions

Reply:
Agreed. 
Also, after the contract isn't in quarantine any more the `delete` would get repeated every time it calls any function with that modifier, which would end up costing more gas than I was trying to save...